src/Device/Driver/AR62xx.cpp

```diff
@@ -0,0 +1,573 @@
+
+  /**
+   * @brief parses the received data
+   * this is needed to save the active and passive station
+   * 
+   * @param string recived data
+   * @param len length of data
+   * @return true if data could be parsed
+   * @return false if data could not be parsed
+   */
+  bool ParseReceivedData(const char *string, size_t len);
+
+  /**
+   * @brief save frequencies
+   * 
+   * @param sz_command command which contains the frequencies
+   * @param len length of the command
+   * @param crc CRC
+   */
+  void SaveAR62xxFrequencies(uint8_t *sz_command, int len, uint16_t crc);
+
+  /**
+   * @brief Get the n-th Byte of an unsigned integer
+   * 
+   * @param number unsigned integer
+   * @param n byte-number
+   * @return uint8_t byte
+   */
+  uint8_t GetNthByte(unsigned number, int n);
+
+  /**
+   * @brief Converts to uint8_t-bytes to an unsigned integer
+   * 
+   * @param first first byte
+   * @param second second byte
+   * @return unsigned unsigned integer
+   */
+  unsigned BytesToUnsigned(uint8_t first, uint8_t second);
+
+public:
+  virtual bool PutActiveFrequency(RadioFrequency frequency,
+                                  const TCHAR *name,
+                                  OperationEnvironment &env) override;
+  virtual bool PutStandbyFrequency(RadioFrequency frequency,
+                                   const TCHAR *name,
+                                   OperationEnvironment &env) override;
+
+  /**
+   * @brief checks if data is received
+   * 
+   * @param _data data which is received
+   * @param length length of the data
+   * @param info nmea info
+   * @return true data has been received
+   * @return false no data has been received
+   */
+  virtual bool DataReceived(const void *data, size_t length, struct NMEAInfo &info) override;
+};
+
+/**
+ * @brief Construct a new AR62xxDevice::AR62xxDevice object
+ * 
+ * @param _port the port to connect to
+ */
+AR62xxDevice::AR62xxDevice(Port &_port) : port(_port) {}
+
+/**
+ * @brief Get the n-th Byte of an unsigned integer
+ * 
+ * @param number unsigned integer
+ * @param n byte-number
+ * @return uint8_t byte
+ */
+uint8_t AR62xxDevice::GetNthByte(unsigned number, int n)
+{
+  return number >> (8 * n) & 0xff;
+}
+
+/**
+ * @brief Converts to uint8_t-bytes to an unsigned integer
+ * 
+ * @param first first byte
+ * @param second second byte
+ * @return unsigned unsigned integer
+ */
+unsigned AR62xxDevice::BytesToUnsigned(uint8_t first, uint8_t second)
+{
+  return second << 8 | first;
+}
+
+/**
+ * @brief checks if data is received
+ * 
+ * @param _data data which is received
+ * @param length length of the data
+ * @param info nmea info
+ * @return true data has been received
+ * @return false no data has been received
+ */
+bool AR62xxDevice::DataReceived(const void *_data, size_t length, struct NMEAInfo &info)
+{
+  /** check we received data */
+  if (_data == nullptr || length <= 0)
```

Comment:
Why is it legal to pass `nullptr` here? Who does it?

Reply:
removed check for nullpointer